src/lib/smartlist_core/smartlist_foreach.h

```diff
@@ -83,6 +83,14 @@
          ++var ## _sl_idx) {                                    \
       var = (sl)->list[var ## _sl_idx];
 
+#define SMARTLIST_FOREACH_REVERSE_BEGIN(sl, type, var)  \
```

Comment:
We should document this new macro and maybe unittest it.

Furthermore, we should document why we are using it in the code instead of the normal SMARTLIST_FOREACH.

Reply:
I did the comment in eb11886fccada2b56d87cf85fda5fb553298bfbf.

I added a note for the unittest in the branch's TODO file, along with notes for a pile of other tests.